### PR TITLE
fix(auth,payments,shared): Block subscription creation when location is not supported according to ToS

### PR DESCRIPTION
**Pull request by xlisachan**
## This pull request

When a country is not a supported location according to our Terms of Service:
  - error message should appear
  - subscription creation is blocked for both credit card and PayPal, if customers are able to get onto the checkout page

## Issue that this pull request solves

Closes: FXA-10602

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

### On Checkout
Screenshot 2024-11-22 at 5 37 03 PM

### After clicking "Subscribe Now"/PayPal buttons
Screenshot 2024-11-21 at 12 56 32 PM

### Storybook - PaymentErrorView
Screenshot 2024-11-21 at 10 55 44 AM

### Storybook - PlanErrorDialog
Screenshot 2024-11-22 at 5 35 26 PM

### CODE REVIEW

File: packages/fxa-shared/lib/errors.ts [hunk 1]
```diff
@@ -123,7 +123,9 @@ export const AUTH_SERVER_ERRNOS = {
 
   UNABLE_TO_LOGIN_NO_PASSWORD_SET: 210,
 
-  SUBSCRIPTION_PROMO_CODE_NOT_APPLIED: 211,
+  SUBSCRIPTION_PROMO_CODE_NOT_APPLIED: 212,
```
Comment (xlisachan) [inline packages/fxa-shared/lib/errors.ts:126]: `211` was already used on L35.

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-auth-server/config/index.ts [hunk 2]
```diff
@@ -158,6 +158,12 @@ const convictConf = convict({
       default: {},
       env: 'GEODB_LOCATION_OVERRIDE',
     },
+    unsupportedLocations: {
+      doc: 'list of unsupported locations according to ToS',
+      default: [],
+      env: 'GEODB_UNSUPPORTED_LOCATIONS',
```
[the remark this answers is not in the document]
Comment (StaberindeZA) [reply packages/fxa-auth-server/config/index.ts:164]: Should the list in the Jira ticket be set as the default?

**Review by StaberindeZA — COMMENTED** [review]
Just some initial very minor notes.

**Review by StaberindeZA — COMMENTED** [review]
I noticed that SubManage also has this error. Is that required/desired as part of this change?

image

I also left a few inline comments with some change requests and a few nits.

nit. I also noticed a different experience between Stripe and Paypal payments. Stripe still had the retry button. I see this as an optional change though, so no need to fix as part of this PR.

Stripe
image


PayPal
image

**Review by StaberindeZA — APPROVED** [review]
r+. ty for figuring this out!

**Pull request merged**